### ISSUE TITLE
application/json Message Replayed as text/plain

### ISSUE THREAD

**Issue by robertbell** [mleech/scotch#2]
Hey - Love the idea for the library and would really like to use it but I've encountered an issue with some initial testing. Not sure whether it's environmental, bug, or a malformed request I'm replaying but the symptoms are this:

I record a HTTP GET request and the content can be deserialized using `HttpResponseMessage.Content.ReadAsAsync<T>` fine. When I replay the same request I get the following exception:

`{"No MediaTypeFormatter is available to read an object of type 'ForexRateResponse' from content with media type 'text/plain'."}`

And the request headers parsed by HttpClient contain two for Content-Type:

```
text/plain; charset=utf-8
application/json; charset=utf-8
```

The json stored in the Cassette file is as follows:

 ```
 {
    "request": {
      "method": "GET",
      "uri": "https://xecdapi.xe.com/v1/convert_from/?from=GBP&to=USD",
      "requestHeaders": [
        {
          "Authorization": "_[redacted]_"
        }
      ],
      "contentHeaders": [],
      "body": ""
    },
    "response": {
      "status": {
        "code": "200",
        "message": "OK"
      },
      "responseHeaders": [
        {
          "Access-Control-Allow-Origin": "*"
        },
        {
          "X-RateLimit-Limit": "900"
        },
        {
          "X-RateLimit-Remaining": "899"
        },
        {
          "X-RateLimit-Reset": "1506360600"
        },
        {
          "Connection": "keep-alive"
        },
        {
          "ETag": "cd47e3f1-12ac-407f-9356-0db7dbb5aa09"
        }
      ],
      "contentHeaders": [
        {
          "Content-Length": "196"
        },
        {
          "Content-Type": "application/json; charset=utf-8"
        },
        {
          "Last-Modified": "Mon, 25 Sep 2017 17:21:00 UTC"
        }
      ],
      "body": "{\"terms\":\"http://www.xe.com/legal/dfs.php\",\"privacy\":\"http://www.xe.com/privacy.php\",\"from\":\"GBP\",\"amount\":1.0,\"timestamp\":\"2017-09-25T17:21:00Z\",\"to\":[{\"quotecurrency\":\"USD\",\"mid\":1.3454740652}]}",
      "httpVersion": "1.1"
    },
    "recordedAt": "2017-09-25T18:21:49.545+01:00"
  }
```

Any idea what may be up? Please let me know if there's any more information I could provide that would be of use!

Thanks in advance,

Rob

**Comment by kevbite** [comment]
Seems the issue is within the `ToHttpResponseMessage` method, It's creating a new instance of a [StringContent](https://msdn.microsoft.com/en-us/library/system.net.http.stringcontent(v=vs.118).aspx) and using that. This Content type will add a default content type header of "text/plain" if not specified (this can be seen here - [StringContent.cs Line 30](https://github.com/mleech/scotch/blob/master/Scotch/HttpInteraction.fs#L97))

You are then later on adding the Content-Type header based on the recorded headers.

I would advise creating your own content type may be `ScotchContent` which derives off `ByteArrayContent` that only takes in the string content. I'd submit a PR but I'm not clever enough to write in fancy languages like F#  :')

**Comment by mleech** [comment]
Thanks @robertbell I'll try to take a look at this soon. And thanks for the pointers @kevbite - that's a big help.

**Comment by mleech** [comment]
I've published version 0.1.1 of the Scotch nuget package with the fix for this in it.
If you get a chance to test it out @robertbell could you let me know if it solves your problem - thanks.

**Comment by robertbell** [comment]
Hey @mleech - thanks for the speedy response (and you too @kevbite!)

Yep issue fixed. In testing I have however spotted another (potential) bug.. I'll open another issue for that.

Thanks again,

Rob
